src/Application.php

```diff
@@ -217,195 +228,50 @@ public function getAdapterFactory()
-
-            // Adapter is not configured ignore
-            if (null === $config) {
-                $ignoredAdapters[] = $adapterName;
-
-                continue;
-            }
 
-            if (!$adapterInfo[AdapterFactory::SUPPORT_REPOSITORY_MANAGER]) {
-                continue;
+            if ($helper instanceof InputAwareInterface) {
+                $helper->setInput($input);
             }
-
-            $adapter = $this->adapterFactory->createRepositoryManager(
-                $adapterName,
-                $config,
-                $this->config
-            );
-
-            if ($adapter->supportsRepository($remoteUrl)) {
-                return $adapter;
-            };
         }
 
-        $exceptionMessage = 'The adapter type could not be determined.';
+        $event = new ConsoleCommandEvent($command, $input, $output);
+        $this->dispatcher->dispatch(ConsoleEvents::COMMAND, $event);
 
-        if ([] !== $ignoredAdapters) {
-            $exceptionMessage .= sprintf(
-                'Note, the following adapters (may support this repository) but are currently not configured: "%s".',
-                implode('", "', $ignoredAdapters)
-            );
+        try {
+            $exitCode = $command->run($input, $output);
+        } catch (\Exception $e) {
+            $event = new ConsoleTerminateEvent($command, $input, $output, $e->getCode());
+            $this->dispatcher->dispatch(ConsoleEvents::TERMINATE, $event);
 
-            $exceptionMessage .= ' Please configure the adapters or run the "core:init" command.';
-        } else {
-            $exceptionMessage .= ' Please run the "core:init" command.';
-        }
+            $event = new ConsoleExceptionEvent($command, $input, $output, $e, $event->getExitCode());
+            $this->dispatcher->dispatch(ConsoleEvents::EXCEPTION, $event);
 
-        throw new \RuntimeException($exceptionMessage);
-    }
+            if ($e instanceof UserException) {
+                $this->getHelperSet()->get('gush_style')->error($e->getMessages());
 
-    /**
-     * Builds the adapter for the application
-     *
-     * @param string|Adapter $adapter
-     * @param array          $config
-     *
-     * @return Adapter
-     *
-     * @throws \RuntimeException when the adapter configuration is invalid
-     */
-    public function buildAdapter($adapter, array $config = null)
-    {
-        if (!$adapter instanceof Adapter) {
-            if (null === $config) {
-                $config = $this->config->get(sprintf('[adapters][%s]', $adapter));
-            }
-
-            if (null === $config) {
-                throw new \RuntimeException(
-                    sprintf(
-                        'The adapter "%s" is not configured yet. Please run the "core:configure" command to configure.',
-                        $adapter
-                    )
-                );
+                if (OutputInterface::VERBOSITY_VERBOSE <= $output->getVerbosity()) {
```

Comment:
This doesn't look very right, if a `UserException` is thrown and verbosity isn't on, then we will never see any output?
Shouldn't any exception always show the error message?